lib/reline/ansi.rb

```diff
@@ -8,6 +8,8 @@ class Reline::ANSI
     'khome' => :ed_move_to_beg,
     'kend'  => :ed_move_to_end,
     'kdch1' => :key_delete,
+    'kpp' => :ed_ignore,
+    'knp' => :ed_ignore,
```

Comment:
It would be better to make it behave the same as IRB using Readline.
```suggestion
    'kpp' => :ed_prev_history,
    'knp' => :ed_next_history,
```

Reply:
Oh, I didn't even think about checking to see what IRB does when using Readline instead of Reline.

@ima1zumi can you tell me how to test IRB with Readline?  Do I just run `irb --nomultiline`?

Because I tried that and PgUp and PgDn did not behave that way.  PgUp got ignored and PgDn turned into a `~`.

Out of curiosity, what shell are you using and how does it behave?

I think you are only seeing this behavior because of your specific setup.  I think maybe your readline has been specifically configured to handle them.  Can you check your `~/.inputrc` and `/etc/inputrc`?  Can you test again after temporarily removing those files or anything else that could be influencing readline?

Reply:
> @sshock I also observed what @ima1zumi described because that's a new behaviour added to `readline 8.2` ([changelog](https://lists.gnu.org/archive/html/info-gnu/2022-09/msg00013.html)). Quote:
> 
> > d. Automatically bind termcap key sequences for page-up and page-down to
> > history-search-backward and history-search-forward, respectively.

Oh, neat!

> I appreciate that you tested readline with IRB yourself and shared your findings. However, it'd be great if you don't make quick assumptions on other people's setup follow by a series of questions. I know the intention was good but it could be seen as a bit aggressive and could make others uncomfortable.

Very sorry.  My system was up-to-date, so it never occurred to me that I might have an old libreadline, and even if I did think of that I never would have imagined that they would begin to provide default behavior for these keys after so many years of not really handling them.

I will try to be more careful in the future and not jump to conclusions.

> Also, a quick way to share setup info is to paste the output of the `irb_info` command:

This is very useful, thanks!

Reply:
> @sshock I also observed what @ima1zumi described because that's a new behaviour added to `readline 8.2` ([changelog](https://lists.gnu.org/archive/html/info-gnu/2022-09/msg00013.html)). Quote:
> 
> > d. Automatically bind termcap key sequences for page-up and page-down to
> > history-search-backward and history-search-forward, respectively.

Good news.  I was able to test on a newer system with readline 8.2 and confirm that pgup/pgdn using IRB w/ readline does indeed have history-search-backward and history-search-forward behavior as described in the changelog.

I will update the PR shortly.  I am in 100% in agreement with making reline match the behavior of readline by default.